chains/evm/calls/transactor/monitored/monitored.go

```diff
@@ -133,52 +158,79 @@ func (t *MonitoredTransactor) Monitor(
 				for k, v := range t.pendingTxns {
 					pendingTxCopy[k] = v
 				}
+				pendingTxTraceIDCopy := make(map[common.Hash]traceapi.TraceID, len(t.pendingTxnsTrace))
+				for k, v := range t.pendingTxnsTrace {
+					pendingTxTraceIDCopy[k] = v
+				}
 				t.txLock.Unlock()
 
 				for oldHash, tx := range pendingTxCopy {
+					if time.Since(tx.submitTime) < tooNewTransaction {
+						continue
+					}
+					tID, ok := pendingTxTraceIDCopy[oldHash]
+					if ok {
+						// Creating span context with existing TraceID
+						spanCtx := traceapi.NewSpanContext(traceapi.SpanContextConfig{TraceID: tID, Remote: true})
+						ctx = traceapi.ContextWithSpanContext(ctx, spanCtx)
+					}
+					ctx, span := otel.Tracer("relayer-sygma").Start(ctx, "relayer.sygma.evm.transactor.Monitor")
+					logger := log.With().Str("dd.trace_id", span.SpanContext().TraceID().String()).Logger()
```

Comment:
This is really Datadog specific and I would remove it.

Reply:
But we are using DataDog, we might just add other attributes in the future if we want to use something else instead of DataDog. 
I guess there might be more abstract way with config to set specific attributes based on trace monitoring tool, but I dp not want to go this way now

Reply:
I am fine with that in relayer, but in core it is unnecesary. And I also don't like this custom logger personally because it is more bloat than worth.

Reply:
We are using many logs from core in sygma, hence we can't ignore them. Moreover if it is not breaking any compatibility while can add some benefits to someone since basically DD is a pretty common tracing tool.